crates/blind_pool/benches/blind_basic.rs

```diff
@@ -0,0 +1,79 @@
+//! Basic benchmarks for the `blind_pool` crate.
+
+#![allow(
+    missing_docs,
+    reason = "No need for API documentation in benchmark code"
+)]
+
+use std::hint::black_box;
+
+use blind_pool::BlindPool;
+use criterion::{Criterion, criterion_group, criterion_main};
+
+criterion_group!(benches, entrypoint);
+criterion_main!(benches);
+
+type TestItem = usize;
+const TEST_VALUE: TestItem = 1024;
+
+fn entrypoint(c: &mut Criterion) {
+    let mut group = c.benchmark_group("bp_fill");
+
+    group.bench_function("empty", |b| {
+        b.iter(|| {
+            drop(black_box(BlindPool::new()));
+        });
+    });
+
+    group.bench_function("one", |b| {
+        b.iter(|| {
+            let mut pool = BlindPool::new();
+            let pooled = pool.insert(TEST_VALUE);
+            pool.remove(pooled);
+            drop(pool);
+        });
+    });
+
+    group.bench_function("ten_thousand", |b| {
+        b.iter(|| {
+            let mut pool = BlindPool::new();
+            let mut pooled_items = Vec::with_capacity(10_000);
+            for _ in 0..10_000 {
+                let pooled = pool.insert(TEST_VALUE);
+                pooled_items.push(pooled);
+            }
+            for pooled in pooled_items {
+                pool.remove(pooled);
+            }
+            drop(pool);
+        });
+    });
+
+    group.finish();
+
+    let mut mixed_group = c.benchmark_group("bp_mixed");
+
+    mixed_group.bench_function("mixed_types", |b| {
```

Comment:
Difficult to understand how this is comparable to anything. Better might be to instead showcase how actions scale with the number of different types. e.g. have a variant of bp_fill that fills the collection with items of 2 different layouts (but still similar enough to be comparable, e.g. 8 bytes u64 plus a custom struct with 4 bytes of data but 8-alignment, to occupy same memory and thereby better control factors to be comparable).